### DISCUSSION TITLE
UnoPLatform Linux set window size programatically

### DISCUSSION THREAD

**Discussion by HowardPWeiss** [unoplatform/uno#19022] [Q&A] [answered]
The MainPage.xaml file in my application sets a Height of 839 and a Width of 1672
When I start my application, I first see a splash page (which is approx 800 x 800) which disappears and then the
main application starts but the window is the same size as the splash screen

Once the application starts, I can resize the window.  This is not what I want

How do I configure my system so the main page comes up in a full size window AND/OR how can I change the window size programatically

``<Page
    x:Class="Hello.Presentation.MainPage"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:Hello.Presentation"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
   
    mc:Ignorable="d"    
    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}"
    Height="839" Width="1672"
    Loaded="OnPageLoad">

    <Page.DataContext>
        <local:MainViewModel/>
    </Page.DataContext>
    <Page.Resources>

**Comment by MartinZikmund** [comment, marked as answer]
You can use the `Window.AppWindow.Resize` API to change the size (you can call it in the `App.xaml.cs` `OnLaunched` method for example). The `Page` is just an element in the UI so it does not influence its parent window.

**Reply by HowardPWeiss** [reply]
Martin

Thank you very much.

Can you point me to an example of using this API in an application running on Linux under visual studio code and written in C#?

Here is the program.cs file which was generated by UNO

```
using Uno.UI.Runtime.Skia;

namespace Hello;

public class Program
{
    [STAThread]
    public static void Main(string[] args)
    {
#if (!useDependencyInjection && useLoggingFallback)
        App.InitializeLogging();

#endif
        var host = SkiaHostBuilder.Create()
            .App(() => new App())
            .UseX11()
            .UseLinuxFrameBuffer()
            .UseMacOS()
            .UseWindows()
            .Build();

        host.Run();
    }
}
```

**Reply by MartinZikmund** [reply]
You can find the Window instance being created in `App.xaml.cs` - you can then call the API above on it
